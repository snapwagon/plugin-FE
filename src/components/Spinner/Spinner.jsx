import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

export default class Spinner extends React.Component {
  calculateStyles() {
    const { size } = this.props;

    if (size < 100) {
      return {
        height: `${size}px`,
        width: `${size}px`,
        borderTop: '3px solid',
        borderRight: '3px solid',
        borderLeft: '3px solid transparent',
        borderBottom: '3px solid transparent'
      };
    }

    return {
      height: `${size}px`,
      width: `${size}px`,
      borderTop: '12px solid',
      borderRight: '12px solid',
      borderLeft: '12px solid transparent',
      borderBottom: '12px solid transparent'
    };
  }

  render() {
    return (
      <i
        style={this.calculateStyles()}
        className={cx(
          'snapW-Spinner',
          this.props.classNames
        )}
      />
    );
  }
}

const {
  bool,
  number,
  objectOf,
  oneOfType,
  string
} = PropTypes;

Spinner.propTypes = {
  classNames: oneOfType([
    string,
    objectOf(bool)
  ]),
  size: number,
};

Spinner.defaultProps = {
  classNames: undefined,
  size: 100,
};
