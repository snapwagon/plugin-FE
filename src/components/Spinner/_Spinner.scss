.snapW-Spinner {
  animation: rotate 1.5s infinite linear;
  border-radius: 100%;
  padding: 2px;
  position: absolute;
  margin-top: -2px;
  margin-left: -2px;
}

@keyframes rotate {
  from { transform: rotate(0deg); }
  to { transform: rotate(359deg); }
}
