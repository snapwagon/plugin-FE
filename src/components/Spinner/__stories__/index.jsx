/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import { storiesOf } from '@storybook/react';

import Spinner from '../Spinner.jsx';

storiesOf('Spinner', module)
  .add('renders the component', () => (
    <div style={{ height: '400px', width: '400px', backgroundColor: '#000' }}>
      <Spinner />
    </div>
  ));
