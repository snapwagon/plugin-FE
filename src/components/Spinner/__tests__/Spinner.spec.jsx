/* eslint-disable no-undef */

import React from 'react';
import { shallow } from 'enzyme';

import Spinner from '../Spinner.jsx';

describe('<Spinner /> snapshots', () => {
  test('renderered as a large <Spinner />', () => {
    const tree = shallow(<Spinner />);
    expect(tree).toMatchSnapshot();
  });

  test('rendered as a small <Spinner />', () => {
    const tree = shallow(<Spinner size={20} />);
    expect(tree).toMatchSnapshot();
  });
});
