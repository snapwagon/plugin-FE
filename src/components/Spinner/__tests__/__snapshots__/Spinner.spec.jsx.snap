// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> snapshots rendered as a small <Spinner /> 1`] = `
<i
  className="pom-Spinner"
  style={
    Object {
      "borderBottom": "3px solid transparent",
      "borderLeft": "3px solid transparent",
      "borderRight": "3px solid",
      "borderTop": "3px solid",
      "height": "20px",
      "width": "20px",
    }
  }
/>
`;

exports[`<Spinner /> snapshots renderered as a large <Spinner /> 1`] = `
<i
  className="pom-Spinner"
  style={
    Object {
      "borderBottom": "12px solid transparent",
      "borderLeft": "12px solid transparent",
      "borderRight": "12px solid",
      "borderTop": "12px solid",
      "height": "100px",
      "width": "100px",
    }
  }
/>
`;
